R/antigen.garnish_run.R

```diff
@@ -171,6 +171,135 @@ ag_data_err <- function(){
+
+  if (!dir.exists(npd)) dir.create(npd)
+
+  # get path to scripts
+  setwd(npd)
+
+  tool_paths <- list.files()
+
+  f <- lapply(tool_paths, function(i){
+
+    # this pattern could one day break.
+    list.files(i, pattern = "netMHC(II)?(pan)?-?([0-9]\\.[0-9])?$",
+        full.names = TRUE)
+
+  }) %>% unlist
+
+  message("Checking netMHC scripts in antigen.garnish data directory.")
+  # sed scripts to correct paths
+  # install data from DTU
+  io <- lapply(f, function(i){
+
+    # rename to take off version, necessary because commands are built into table
+    # in get_pred_commands and check_pred_tools hasn't run at that point
+    io <- file.path(dirname(i),
+    basename(i) %>% stringr::str_replace("-.*$", ""))
+
+    # check if scripts were already edited
+    if (file.exists(file.path(dirname(i), "itwasedited.txt")))
+      return(io)
+
+    # formatting for these links is  not consistent,  try no uname first
+    # netMHC is always capitalized in the links it seems
+    # dirname(i) will always have version number
+    link_to_data <- paste0("http://www.cbs.dtu.dk/services/",
+        dirname(i) %>% stringr::str_replace("net", "Net"),
+        "/data.tar.gz")
+
+    cmd <- paste("curl -fsSL", link_to_data, ">", "dtu.tar.gz")
+
+    suppressWarnings({
+    curl_status <- system(cmd, intern = TRUE)
+    })
+
+    if (!is.null(attr(curl_status, "status")) &&
+        attr(curl_status, "status") == 22){
+
+      un <- system("uname -s", intern = TRUE)
+
+      if (!un %chin% c("Linux")) stop("Linux only.")
+
+      cmd <- cmd %>%
+      stringr::str_replace("data\\.tar\\.gz", paste0("data.", un, ".tar.gz"))
+
+      message(paste("Downloading data for", dirname(i), "on", un))
+
+      suppressWarnings({
+      curl_status <- system(cmd, intern = TRUE)
+      })
+
+    }
+
+    if (!file.exists("dtu.tar.gz"))
+    stop(paste("Unable to download data tar from DTU. See ReadMe in", file.path(dir, "netMHC", dirname(i))))
+
+    # move it into the right folder and untar
+    dtar <- file.path(dirname(i), "dtu.tar.gz")
+
+    file.rename("dtu.tar.gz", dtar)
+
+    setwd(dirname(i))
+
+    system(paste("tar -xzvf", basename(dtar)))
+
+    setwd(npd)
+
+    line <- file.path(dir, "netMHC", dirname(i))
+
+    line <- paste("setenv NMHOME ", line %>% stringr::str_replace_all("/", "\\\\/"), sep = "")
+    line2 <- paste("setenv TMPDIR ", "$HOME", sep = "")
+
+    cmd <- paste("cat ", i, " | ",
+    # place holders for variable references
+    "sed 's/\\$NMHOME/placeholderxxx/' | ",
+    "sed 's/\\$TMPDIR/placeholderyyy/' | ",
+    "sed 's/setenv.*NMHOME.*$/", line, "/' | ",
+    # reset placeholder
+    "sed 's/placeholderxxx/$NMHOME/' | ",
+    "sed 's/setenv.*TMPDIR.*$/", line2, "/' | ",
+    # reset placeholder
+    "sed 's/placeholderyyy/$TMPDIR/' > ", "placeholder_script.txt", sep = "")
+
+    system(cmd)
+
+    # add marker that it was edited
+    system(paste("touch", file.path(dirname(i), "itwasedited.txt")))
```

Comment:
instead of this system that could break, you could save the original files as .bk first, then always copy these .bk files to make new, edited files with the correct patterns. then, if your txt here is moved or something is wrong, running the script again always works.

Reply:
good call, no checks on if it was edited that way.